### ISSUE TITLE
coherent.deps.pypi.NoDistributionForImport: _typeshed.StrPath

### ISSUE THREAD

**Issue by jaraco** [coherent-oss/coherent.test#15]
Attempting to build the project emits:

```
coherent.deps.pypi.NoDistributionForImport: _typeshed.StrPath
```

**Comment by bswck-hai** [comment]
Yup, that was my change. I added a special import of a symbol from `_typeshed` typing-only stable API.

**Comment by bswck-hai** [comment]
If that's a big issue for now, you could simply use `str | os.PathLike[str]` for now instead.

**Comment by bswck-hai** [comment]
Oh, sorry. Writing from my work account.

**Comment by jaraco** [comment]
Indeed, that root module doesn't show up in the database.

```
 🐚 http https://us-east-1.aws.data.mongodb-api.com/app/application-0-ponwfin/endpoint/distributions_for root==_typeshed
HTTP/1.1 200 OK
content-encoding: gzip
content-length: 28
content-type: application/json
date: Sat, 21 Sep 2024 19:21:58 GMT
server: mdbws
strict-transport-security: max-age=31536000; includeSubdomains;
vary: Origin
x-appservices-request-id: 66ef1cd5f0b732b4ec0cb426
x-frame-options: DENY
x-xgen-up-proto: HTTP/2

null
```

Who supplies `_typeshed`?

**Comment by bswck-hai** [comment]
Type checkers. Mypy and pyright see `_typeshed` in the typeshed because it comes bundled with them. Maybe we could somehow specialcase those, using https://github.com/python/typeshed.

**Comment by jaraco** [comment]
Perhaps `if type_checking` should be excluded when inferring the imports, but that's a feature not yet implemented. I'd like to handle that feature alongside environment-specific checks (e.g. `if sys.version_info <` or `if plaform.system() == `), and include the TYPE_CHECKING dependencies in the `type` extra. For now, maybe `coherent.deps` should ignore that one as a special case.

**Comment by bswck** [comment]
Exactly. I think the same. 👍
